### PR TITLE
using self._waiting

### PR DESCRIPTION
**Pull request by xiaoshuai**

### CODE REVIEW

File: pyee/asyncio.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def _emit_run(
                 return
 
             def callback(f):
-                self._waiting.remove(f)
+                self._waiting.discard(f)
```
Comment (jfhbrook) [inline pyee/asyncio.py:70]: What does this change do?

**Review by jfhbrook — COMMENTED** [review]
This looks like a good idea. I'd love if you can put down a PR description! Docstrings would be good too. Full disclosure, it will be some time before I can review and integrate this fully.

File: pyee/asyncio.py (hunk 1 above)
Comment (xiaoshuai) [reply pyee/asyncio.py:70]: When we call the `cancel_all` function, the `_waiting` set has already been emptied.
the `remove` raises a `KeyError` exception, while `discard` does nothing.

**Comment by xiaoshuai** [comment]
> This looks like a good idea. I'd love if you can put down a PR description! Docstrings would be good too. Full disclosure, it will be some time before I can review and integrate this fully.

sorry, my english is poor.

File: pyee/asyncio.py (hunk 1 above)
Comment (jfhbrook) [reply pyee/asyncio.py:70]: Ah, that makes sense.

**Comment by jfhbrook** [comment]
> sorry, my english is poor.

Hey, no problem. I understand what you're trying to do here.

I took and ran with this PR in #168. I'll go ahead and merge that one, and close this one.

Thanks for your great idea!

**Comment by jfhbrook** [comment]
@xiaoshuai The code in #168 is released in v12.1.1. You can read the updated documentation [here](https://pyee.readthedocs.io/en/latest/api/#pyee.asyncio.AsyncIOEventEmitter).

Cheers!

**Pull request closed without merging**